convert/src/label_studio_to_badgerdoc/label_studio_to_badgerdoc_use_case.py

```diff
@@ -0,0 +1,431 @@
+                importjob_id_created, file_id_in_assets
+            )
+            self.s3_client.upload_file(
+                str(badgerdoc_annotations_path),
+                self.s3_output_bucket,
+                s3_output_annotations_path,
+            )
+
+    def request_jobs_to_create_annotation_job(
+        self,
+        file_id_in_assets,
+        owner,
+        validation_type,
+        deadline,
+        extensive_coverage,
+        annotators,
+        validators,
+    ) -> int:
+        categories = self.request_annotations_to_post_categories()
+        post_annotation_job_url = f"{settings.job_service_url}create_job/"
+        post_annotation_job_body = {
+            "name": f"import_label_studio_job_{uuid4()}",
+            "type": "AnnotationJob",
+            "files": [file_id_in_assets],
+            "datasets": [],
+            "is_draft": False,
+            "validation_type": validation_type,
+            "annotators": annotators,
+            "owners": [owner],
+            "categories": categories,
+            "validators": validators,
+        }
+        if deadline:
+            post_annotation_job_body.update(
+                {"deadline": jsonable_encoder(deadline)}
+            )
+        if extensive_coverage is not None:
+            post_annotation_job_body.update(
+                {"extensive_coverage": extensive_coverage}
+            )
+        LOGGER.debug(
+            "Making a request to create an Annotation Job in 'jobs' to url: %s with request body: %s",
+            post_annotation_job_url,
+            post_annotation_job_body,
+        )
+
+        try:
+            request_to_post_annotation_job = requests.post(
+                url=post_annotation_job_url,
+                headers=self.request_headers,
+                json=post_annotation_job_body,
+            )
+            request_to_post_annotation_job.raise_for_status()
+        except requests.exceptions.RequestException as e:
+            LOGGER.exception(
+                "Failed request to 'jobs' to post AnnotationJob - %s", e
+            )
+            raise HTTPException(
+                status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+                detail=f"Failed request to 'jobs' to post AnnotationJob",
+            ) from e
+
+        LOGGER.debug(
+            "Got this response from jobs service: %s",
+            request_to_post_annotation_job.json(),
+        )
+        return request_to_post_annotation_job.json()["id"]
+
+    def get_categories_of_links(self, pages_objs):
+        result = []
+        for pages_obj in pages_objs:
+            for link in pages_obj.links:
+                result.append(link.category_id)
+
+        return result
+
+    def form_post_category_body(self, type_of_categories, category):
+        possible_categories_colors = [
+            "red",
+            "blue",
+            "green",
+            "purple",
+            "orange",
+            "grey",
+        ]
```

Comment:
Tuple would work fine for it

Reply:
agreed

Reply:
You made these colors in hexadecimal format, would be nice to have at least a comment what color it is. Or better move them to vars.